crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/rpc.rs

```diff
@@ -31,9 +31,19 @@ pub struct UsedResources {
 impl UsedResources {
     pub fn extend(self: &mut UsedResources, other: &UsedResources) {
         self.execution_resources.vm_resources += &other.execution_resources.vm_resources;
-        self.execution_resources
+
+        other
+            .execution_resources
             .syscall_counter
-            .extend(&other.execution_resources.syscall_counter);
```

Comment:
Funny, this caused syscalls to not accumulate, as `map1.extend(map2)` replaced already existing entries in `map1` with entries in `map2`